packages/console/src/browser/console-widget.ts

```diff
@@ -295,4 +311,23 @@ export class ConsoleWidget extends BaseWidget implements StatefulWidget {
         return this._input && this._input.isFocused({ strict: true });
     }
 
+    // To set the active language for the console input text model.
+    // https://github.com/microsoft/vscode/blob/2af422737386e792c3fcde7884f9bf47a1aff2f5/src/vs/workbench/contrib/debug/browser/repl.ts#L371-L384
+    protected setMode(): void {
+        if (this.isHidden) {
+            return;
+        }
+
+        const activeEditorControl = this.editorService.getActiveCodeEditor();
+        if (activeEditorControl) {
+            this._modelChangeListener.dispose();
+            this._modelChangeListener = activeEditorControl.onDidChangeModelLanguage(() => this.setMode());
```

Comment:
Suggestion: This new `_modelChangeListener` does not get added to the `toDispose` list. Instead, we need to manually dispose this element in our own `dispose` method of this widget.

Reply:
Thank you! Fixed via https://github.com/eclipse-theia/theia/pull/14854/commits/104fe67f91fee6baa24536920fd7a12689f156bf.